subsys/bluetooth/host/smp.c

```diff
@@ -2721,6 +2721,10 @@ static uint8_t remote_sec_level_reachable(struct bt_smp *smp)
 {
 	struct bt_conn *conn = smp->chan.chan.conn;
 
+	if (get_encryption_key_size(smp) < BT_SMP_MAX_ENC_KEY_SIZE &&
+		IS_ENABLED(CONFIG_BT_SMP_SC_ONLY)) {
```

Comment:
The indentation isn't quite right here. Also, I find it more natural to have the `IS_ENABLED() as the first condition since that's kind of the key to even considering any of the other conditions.

Reply:
Actually, is this entire addition even needed? Isn't it covered by 'case BT_SECURITY_L4' further down in the function?

Reply:
Ah I see - I was confusing the suggestion from @joerchan with the actual chances (they look very similar in the GitHub UI). Please follow what @joerchan suggested - that looks the cleanest to me.

Reply:
@jhedberg Only when we have initiated security ourselves, in this case the peer is initiating, and we have enabled SC_ONLY.
My suggestion should handle this case.

Reply:
Ok, I updated PR as @joerchan described